test/e2e/k8s_events_test.go

```diff
@@ -0,0 +1,106 @@
+// +build e2e
+
+/*
+Copyright 2018 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"testing"
+	"time"
+
+	"github.com/knative/eventing/test"
+	pkgTest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+)
+
+const (
+	serviceAccount   = "e2e-receive-adapter"
+	eventSource      = "k8sevents"
+	routeName        = "e2e-k8s-events-function"
+	channelName      = "e2e-k8s-events-channel"
+	provisionerName  = "in-memory-channel"
+	subscriptionName = "e2e-k8s-events-subscription"
+)
+
+func TestKubernetesEvents(t *testing.T) {
+
+	logger := logging.GetContextLogger("TestKubernetesEvents")
+
+	clients, cleaner := Setup(t, logger)
+
+	test.CleanupOnInterrupt(func() { TearDown(clients, cleaner, logger) }, logger)
```

Comment:
What does this do that the `defer` doesn't?

Reply:
Nothing afaict. @vaikas-google why is this here along with a `defer(TearDown)`?

Reply:
my bad probably :)